packages/rulesets/src/asyncapi/functions/asyncApi2DocumentSchema.ts

```diff
@@ -89,24 +101,51 @@ function getSchema(formats: Set<Format>): Record<string, unknown> | void {
   }
 }
 
-export default createRulesetFunction<unknown, null>(
+// For optimizing the retrieving/creation of AJV's validation funcition for a given AsyncAPI version.
+// Currently each validation run creates a separate `documentInventory` (needed by `schemaFn`), which serves as an identifier for the weakMap's element of available AJV's validation functions.
+// This variable will always be the same for each validation run, regardless of the number of runs or Spectral instances.
+const CONST_DOCUMENT_INVENTORY: RulesetFunctionContext['documentInventory'] =
+  {} as RulesetFunctionContext['documentInventory'];
```

Comment:
There I have big problem. Due to fact that `schema` function from `@stoplight/spectral-functions` uses the context's `documentInventory` reference to save and reuse AJV's validation function for given schema I made that above trick to optimize validation. Should we add option to the `schema` function to allow above logic in a simpler way? 

cc @P0lip

Reply:
Agreed, that'd be a nice addition